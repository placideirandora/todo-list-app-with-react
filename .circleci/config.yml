version: 2.1
jobs:
  build:
    docker:
      - image: circleci/node:10.16.0

    working_directory: ~/todo-list-app-with-react-js

    steps:
      - checkout
      - restore_cache:
          keys:
            - v1-dependencies-{{ checksum "package.json" }}
            - v1-dependencies-

      - run: npm install

      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-{{ checksum "package.json" }}

      - run: npm run test:coverage
