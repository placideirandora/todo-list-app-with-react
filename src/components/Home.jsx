import React, { Fragment } from 'react';
import { homeStyles } from '../styles/home';
import todoSvg from '../assets/images/todo';
import Image from './shared/Image';

const Home = () => {
  return (
    <Fragment>
      <section style={homeStyles.homeWrapper}>
        <Image source={todoSvg} />
        <p>Manage your todos in an effective and efficient manner!</p>
      </section>
    </Fragment>
  );
};

export default Home;
