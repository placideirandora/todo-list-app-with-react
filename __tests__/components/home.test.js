import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as adapter from '../config/enzyme.config';
import Home from '../../src/components/Home';

describe('Home Component Test Cases', () => {
  it('should render the component without any failure', () => {
    const component = shallow(<Home />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
