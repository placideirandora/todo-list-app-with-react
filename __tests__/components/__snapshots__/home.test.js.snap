// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Test Cases should render the component without any failure 1`] = `
<Fragment>
  <section
    style={
      Object {
        "fontFamily": "Patua One",
        "fontSize": "2.6rem",
        "letterSpacing": "0.03rem",
        "paddingTop": "2rem",
        "textAlign": "center",
        "textTransform": "capitalize",
      }
    }
  >
    <Image
      height={300}
      source="todo.svg"
      width={400}
    />
    <p>
      Manage your todos in an effective and efficient manner!
    </p>
  </section>
</Fragment>
`;
